Source/UI/Validator/PhoneNumberValidator.swift

```diff
@@ -1,11 +1,39 @@
 import Foundation
+import PhoneNumberKit
+import Checkout
 
 class PhoneNumberValidator: Validator {
     private let validator = PhoneValidator()
+    private let phoneKit = PhoneNumberKit()
+    private let validCharacterSet: CharacterSet = {
+        var validInputs = "+ "
+        (0...9).forEach { validInputs.append("\($0)") }
+        return CharacterSet(charactersIn: validInputs)
+    }()
 
-    func validate(value: Any?) -> Bool {
-        guard let phoneNumber = (value as? Phone) else { return false }
-        return validator.validate(phoneNumber) == .success
+    func shouldAccept(text: String) -> Bool {
+        CharacterSet(charactersIn: text).isSubset(of: validCharacterSet) &&
+            text.count < Checkout.Constants.Phone.phoneMaxLength
+    }
+
+    func isValid(text: String) -> Bool {
+        do {
+            let formattedNumber = try phoneKit.parse(text, ignoreType: true)
```

Comment:
question what is the phoneKit.parse do ? here. Space formatting of phone numbers.

Reply:
`phoneKit.parse` uses PhoneNumberKit to parse a String into a phone number. If it contains country code it uses the expected format for that country, if not it defaults to device locale (as we did previously)

_reference [PhoneNumberKit repo](https://github.com/marmelroy/PhoneNumberKit#usage)_